freedomofpress/securedrop-client
# Supports opening submissions in DispVMs from Qubes dev env

# Description

Adds support for running the client code (via `./run.sh`) within a Qubes AppVM, and opening submissions in DispVMs from the client interface. Related issues:

* Closes https://github.com/freedomofpress/securedrop-client/issues/482
* Closes https://github.com/freedomofpress/securedrop-client/issues/483
* Refs https://github.com/freedomofpress/securedrop-workstation/issues/288
* Refs https://github.com/freedomofpress/securedrop-debian-packaging/pull/72

# Test Plan

Throughout these instructions, `sd-dev` is assumed to be the name of the developer's AppVM where code is written. If you use a different VM name, substitute that below.

* Clone/checkout https://github.com/freedomofpress/securedrop-workstation/pull/299 and run `make all`
* In `dom0`, run `qvm-tags sd-dev add sd-client`
* In `sd-dev`, clone/checkout the "securedrop" repository and run `make dev` to start the containers
* In `sd-dev`, clone/checkout this PR branch, and run `./run.sh` to start the client interface
* Submit a PDF to the Source Interface, confirm visible in the Client
* Open the PDF submission in the Client, confirm it opens in a DispVM
* Send a reply to a source, confirm no errors


# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs, network (via the RPC service) traffic, or fine tuning of the graphical user interface, Qubes testing is required. Please check as applicable:
